permissions/resource_permissions.go

```diff
@@ -215,6 +215,16 @@ func (a PermissionsAPI) Delete(objectID string) error {
 				accl.AccessControlList = append(accl.AccessControlList, change)
 			}
 		}
+
+		// handle special case when we add extra permission to a user home dir
+		if v, ok := d.GetOk("directory_path"); ok {
+			if v.(string) == fmt.Sprintf("/Users/%s", acl.UserName) && acl.AllPermissions[0].PermissionLevel == "CAN_MANAGE" {
```

Comment:
Why do we only compare with the first index of the list? Just to confirm, is it necessar that CAN_MANAGE will always be at 0 index?

Reply:
@tanmay-db That is... a great question, considering that this PR has been hanging here for a while - I'm not sure... I'm unfortunately not that knee-deep in databricks provider code, would you mind suggesting an alternative?

Reply:
@alexott and me discussed this and we think going ahead with checking all the items in the list would be more safer.

Reply:
Just to check, after this extending this to SPN and adding tests would be left before we can merge this right?

Reply:
@tanmay-db Something like this?

Reply:
> Just to check, after this extending this to SPN and adding tests would be left before we can merge this right?

Expanding to SPN homedir you mean? Yeah, I'll need to work on that too, although I have no ideas what the format of SPN's home dir is... is it `/Users/<spn_id>`? If so, which ID is it, Azure Object ID, Azure app ID or Databricks internal ID? :)

As for tests... I'll do my best, but would really appreciate some help, I'm incredibly new to your codebase :)

Reply:
it's SPN's Application ID (UUID)

Reply:
@favoretti, yes, to optimise a little, we can add break in the loop when we find the permission level to CAN_MANAGE.